bool collision
float64 time_elapsed
bool new_run
bool end_of_action
float64 action_progress